src/Utilities/PooledMemory.h

```diff
@@ -0,0 +1,244 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Jeongnim Kim, jeongnim.kim@intel.com, Intel Corp.
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_POOLEDMEMORY_H
+#define QMCPLUSPLUS_POOLEDMEMORY_H
+
+#include <vector>
+#include <complex>
+#include <limits>
+#include "simd/allocator.hpp"
+
+namespace qmcplusplus
+{
+
+/** Memory pool to manage arrays+scalars with alignment
+ * @tparam T_scalar : type of the scalar data type, typically OHMMS_PRECISION_FULL
+ * @tparam PageSize : page size in bytes, default=4096 (4K)
+ * @tparam Alloc : allocator, Mallocator<T, PageSize>
+ *
+ * The bulk part is accessed directly by address
+ * The scalar part works as PooledData, all the values are static_cast to T_scalar.
+ */
+template<typename T_scalar=OHMMS_PRECISION_FULL,
+         size_t PageSize=4096,
+         typename Alloc=Mallocator<char, PageSize> >
+struct PooledMemory
```

Comment:
There should be a unit test added for this class.

Reply:
I agree. Do you have time to help me on making the unit tests?

Reply:
Yes, I can help with making some unit tests.